matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/MatrixPatterns.kt

```diff
@@ -165,6 +165,8 @@ object MatrixPatterns {
     fun candidateAliasFromRoomName(name: String): String {
         return Regex("\\s").replace(name.lowercase(), "_").let {
             "[^a-z0-9._%#@=+-]".toRegex().replace(it, "")
+        }.let { alias ->
+            if (alias.length > 255) alias.substring(0, 255) else alias
```

Comment:
I would simplify by always taking the substring here, no need too check the length.

Reply:
also deduce the domain length + 1 (for `#`)